### PR TITLE
Mention that the traits in `convert` are internal

### LINKED ISSUE #4023: Document more prominently that traits like `TryFromJsValue` are for internal use only
I was reviewing code using [`wasm_bindgen::convert::TryFromJsValue`](https://docs.rs/wasm-bindgen/0.2.92/wasm_bindgen/convert/trait.TryFromJsValue.html) and it wasn't obvious to me that this is not meant to be used unless you know what you're doing. Could this be documented more clearly? It's already documented on the [`wasm_bindgen::convert`](https://docs.rs/wasm-bindgen/0.2.92/wasm_bindgen/convert/) module, but I didn't stumble upon this in the code review.

Some options:
1) rename the `convert` module to `internal` to make it more obvious from the import path.
2) mention that it's an internal item on all the items in the `convert` module.
3) mark the module as `#[doc(hidden)]` so they're not easily found when trying to look through the documentation.

### PR DESCRIPTION
**Pull request by hrxi**
Fixes #4023.

### CODE REVIEW

**Comment by hrxi** [comment]
Test failure looks unrelated.

File: src/convert/traits.rs [hunk 1]
```diff
@@ -4,10 +4,14 @@ use core::ops::{Deref, DerefMut};
 use crate::describe::*;
 use crate::JsValue;
 
-/// A trait for anything that can be converted into a type that can cross the
-/// wasm ABI directly, eg `u32` or `f64`.
+/// An [**internal**](crate::convert) trait for anything that can be converted
+/// into a type that can cross the wasm ABI directly, eg `u32` or `f64`.
```
[the remark this answers is not in the document]
Comment (hrxi) [reply src/convert/traits.rs]: Done.

File: src/convert/mod.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-//! This is mostly an internal module, no stability guarantees are provided. Use
-//! at your own risk.
+//! This is mostly an internal module, no stability guarantees are provided.
+//! Only use if you know what you are doing.
```
[the remark this answers is not in the document]
Comment (hrxi) [reply src/convert/mod.rs]: Done.

File: src/convert/mod.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
-//! This is mostly an internal module, no stability guarantees are provided. Use
-//! at your own risk.
+//! This is mostly an internal module, no stability guarantees are provided.
```
[the remark this answers is not in the document]
Comment (hrxi) [reply src/convert/mod.rs]: Done.

**Comment by hrxi** [comment]
All feedback addressed, ready for review.

**Review by daxpedda — APPROVED** [review]
Thank you!

**Pull request merged**